File: backend/src/main/java/com/desafio/backend/service/ServidorService.java
package com.desafio.backend.service;

import com.desafio.backend.dto.ServidorRequestDTO;
import com.desafio.backend.model.Servidor;
import java.util.List;

public interface ServidorService {
    List<Servidor> listarTodos();
    Servidor salvar(ServidorRequestDTO dto);
    Servidor atualizar(Long id, ServidorRequestDTO dto);
    void deletar(Long id);
}

File: frontend/src/app/core/services/servidor.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Servidor } from '../models/servidor.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ServidorService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/servidores`;

  listar(): Observable<Servidor[]> {
    return this.http.get<Servidor[]>(this.apiUrl);
  }

  salvar(servidor: Servidor): Observable<Servidor> {
    return this.http.post<Servidor>(this.apiUrl, servidor);
  }

  atualizar(id: number, servidor: Servidor): Observable<Servidor> {
    return this.http.put<Servidor>(`${this.apiUrl}/${id}`, servidor);
  }

  deletar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
